ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/TeamCreationPanel.java

```diff
@@ -64,67 +64,86 @@ public void setPieceType(PieceType pieceType) {
-
-            JPanel teamNamePanel = new JPanel();
-            teamNamePanel.setOpaque(false);
-            teamNamePanel.setLayout(new BoxLayout(teamNamePanel, BoxLayout.LINE_AXIS));
-            teamNamePanel.add(GuiUtility.createJLabel(Messages.getString("TeamCreationPanel.name")));
-
-            Dimension minSize = new Dimension(5, 25);
-            Dimension prefSize = new Dimension(5, 25);
-            Dimension maxSize = new Dimension(Short.MAX_VALUE, 25);
-            teamNamePanel.add(new Box.Filler(minSize, prefSize, maxSize));
-
-            teamNamePanel.add(Box.createHorizontalGlue());
-
-            mTeamNameFields[offsetTeamId] = new JTextField(20);
-            mTeamNameFields[offsetTeamId].setText(teamInfo.teamName);
-            mTeamNameFields[offsetTeamId].setPreferredSize(new Dimension(50, 25));
-            teamNamePanel.add(mTeamNameFields[offsetTeamId]);
-
-            teamPanel.add(new Box.Filler(new Dimension(5, 10), new Dimension(5, 10), new Dimension(Short.MAX_VALUE, 10)));
-            teamPanel.add(teamNamePanel);
-
-            JPanel teamColorPanel = new JPanel();
-            teamColorPanel.setOpaque(false);
-            teamColorPanel.setLayout(new BoxLayout(teamColorPanel, BoxLayout.LINE_AXIS));
-            teamColorPanel.add(GuiUtility.createJLabel(Messages.getString("TeamCreationPanel.color")));
-
-            teamColorPanel.add(new Box.Filler(minSize, prefSize, maxSize));
-            teamColorPanel.add(Box.createHorizontalGlue());
-
-            mTeamColorFields[offsetTeamId] = new JTextField(20);
-            mTeamColorFields[offsetTeamId].setBackground(teamInfo.color);
-            mTeamColorFields[offsetTeamId].setEditable(false);
-            createMouseListener(mTeamColorFields[offsetTeamId], teamInfo);
-            teamColorPanel.add(mTeamColorFields[offsetTeamId]);
-
-            teamPanel.add(new Box.Filler(new Dimension(5, 10), new Dimension(5, 10), new Dimension(Short.MAX_VALUE, 10)));
-            teamPanel.add(teamColorPanel);
-
-            constraints.gridx = teamInfo.teamId;
-            constraints.gridy = 0;
-            constraints.weightx = 1.0;
-            constraints.weighty = 1.0;
-            constraints.insets = new Insets(0, 25, 10, 25);
-            constraints.anchor = GridBagConstraints.CENTER;
-            constraints.fill = GridBagConstraints.BOTH;
-            add(teamPanel, constraints);
+            createTeamPanel(squareConfig, teamInfo);
+        });
+        JButton addTeam = new JButton(Messages.getString("TeamCreationPanel.addTeam"));
+        addTeam.addActionListener(event -> {
+            TeamInfo newTeam = new TeamInfo();
+            newTeam.teamId = mTeamIDCounter;
+            newTeam.color = Color.gray;
+            newTeam.teamName = Messages.getString("TeamCreationPanel.team", mTeamIDCounter++);
+            mTeamInfos.add(newTeam);
+            createTeamPanel(squareConfig, newTeam);
+            remove(addTeam);
+            if (mTeamIDCounter <= 4)
```

Comment:
Does it work to only remove the addTeam item if we now have 4 views? Or do we have to remove and re-add it?